src/librustc_passes/region.rs

```diff
@@ -720,6 +720,7 @@ impl<'tcx> Visitor<'tcx> for RegionResolutionVisitor<'tcx> {
         let outer_ec = mem::replace(&mut self.expr_and_pat_count, 0);
         let outer_cx = self.cx;
         let outer_ts = mem::take(&mut self.terminating_scopes);
+        let outer_pessimistic_yield = mem::replace(&mut self.pessimistic_yield, false);
```

Comment:
Could you add a comment linking to #69307?

Reply:
done